composeApp/src/commonMain/kotlin/sp/bvantur/tasky/event/ui/SingleInputScreen.kt

```diff
@@ -0,0 +1,135 @@
+package sp.bvantur.tasky.event.ui
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBackIos
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import org.koin.compose.viewmodel.koinViewModel
+import org.koin.core.parameter.parametersOf
+import sp.bvantur.tasky.core.ui.theme.appBarAction
+import sp.bvantur.tasky.core.ui.utils.CollectSingleEventsWithLifecycle
+import sp.bvantur.tasky.event.presentation.SingleInputSingleEvent
+import sp.bvantur.tasky.event.presentation.SingleInputViewModel
+import sp.bvantur.tasky.event.presentation.SingleInputViewState
+import sp.bvantur.tasky.event.ui.model.SingleInputModel
+
+@Composable
+fun SingleInputRoute(
+    singleInputModel: SingleInputModel?,
+    onNavigateBack: () -> Unit,
+    onSaveAction: (String, Boolean) -> Unit
+) {
+    val viewModel = koinViewModel<SingleInputViewModel>(
+        parameters = { parametersOf(singleInputModel) }
+    )
```

Comment:
Same question here with regards to accessing this data directly in your viewmodel by using your `SavedStateHandle`. Koin automatically handles its injection when you use `viewModelOf()` in your DI `module`